pkg/build/api/v1/types.go

```diff
@@ -477,6 +477,13 @@ type SourceBuildStrategy struct {
 
 	// forcePull describes if the builder should pull the images from registry prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// RuntimeImage is the optinal image that is used to run an application
+	// without unneeded dependencies installed.
+	RuntimeImage *kapi.ObjectReference `json:"runtimeImage,omitempty"`
+
+	// RuntimeArtifacts holds the mapping between the artifacts paths on the builder image and the paths on the runtime image where they should be copied.
+	RuntimeArtifacts []ImageSourcePath `json:"runtimeArtifacts,omitempty"`
```

Comment:
> Doesn't download builder image on the node where the build gets executed.

well the s2i code that's being executed by openshift does pull the builder image down if either:
1) the pull policy is pull always/forcepull
2) the image is not present on the node

at which point it uses the pullsecret on the buildconfig, if one is defined.  if one is not defined, it uses the service account token (which will work for pulling from the openshift cluster registry, but obviously not dockerhub/etc)